src/storage.rs

```diff
@@ -492,7 +479,9 @@ impl Storage for MemStorage {
         } else {
             let mut snap = core.snapshot();
             if snap.get_metadata().index < request_index {
+                // NOTE: the logic is only for tests.
                 snap.mut_metadata().index = request_index;
+                snap.mut_metadata().conf_state_index = request_index;
```

Comment:
This is confusing. What if there are conf change between `conf_state_index` and `request_index`?